crates/router/src/db/health_check.rs

```diff
@@ -0,0 +1,157 @@
+                        errors::HealthCheckDBError::DBReadError
+                    })?;
+
+                    logger::debug!("Database read was successful");
+
+                    db.insert_config(ConfigNew {
+                        key: "test_key".to_string(),
+                        config: "test_value".to_string(),
+                    })
+                    .await
+                    .map_err(|err| {
+                        logger::error!(write_err=?err,"Error while writing to database");
+                        errors::HealthCheckDBError::DBWriteError
+                    })?;
+
+                    logger::debug!("Database write was successful");
+
+                    db.delete_config_by_key("test_key").await.map_err(|err| {
+                        logger::error!(delete_err=?err,"Error while deleting element in the database");
+                        errors::HealthCheckDBError::DBDeleteError
+                    })?;
+
+                    logger::debug!("Database delete was successful");
+
+                    Ok::<_, errors::HealthCheckDBError>(())
+                })
+            })
+            .await?;
+
+        Ok(())
+    }
+
+    async fn health_check_redis(
+        &self,
+        db: &dyn StorageInterface,
+    ) -> CustomResult<(), errors::HealthCheckRedisError> {
+        let redis_conn = db
+            .get_redis_conn()
+            .change_context(errors::HealthCheckRedisError::RedisConnectionError)?;
+
+        redis_conn
+            .serialize_and_set_key_with_expiry("test_key", "test_value", 30)
+            .await
+            .change_context(errors::HealthCheckRedisError::SetFailed)?;
+
+        logger::debug!("Redis set_key was successful");
+
+        redis_conn
+            .get_key("test_key")
+            .await
+            .change_context(errors::HealthCheckRedisError::GetFailed)?;
+
+        logger::debug!("Redis get_key was successful");
+
+        redis_conn
+            .delete_key("test_key")
+            .await
+            .change_context(errors::HealthCheckRedisError::DeleteFailed)?;
+
+        logger::debug!("Redis delete_key was successful");
+
+        Ok(())
+    }
+
+    async fn health_check_locker(
+        &self,
+        state: &routes::AppState,
+    ) -> CustomResult<u16, errors::HealthCheckLockerError> {
+        let locker = &state.conf.locker;
+        let mut status_code = 0;
+        if !locker.mock_locker {
+            let mut url = locker.host_rs.to_owned();
+            url.push_str("/health");
+            let request = services::Request::new(services::Method::Get, &url);
+            status_code = services::call_connector_api(state, request)
+                .await
+                .change_context(errors::HealthCheckLockerError::FailedToCallLocker)?
+                .map(|resp| resp.status_code)
+                .map_err(|err| err.status_code)
+                .unwrap_or_else(|code| code);
+        }
```

Comment:
Include the path into the exclude proxy list. Also, can we define the path as a separate constant?

Reply:
all the paths in `proxy_bypass_urls`?